WooCommerce/Classes/POS/Controllers/PointOfSaleCouponsController.swift

```diff
@@ -64,20 +66,53 @@ import protocol Yosemite.PointOfSaleCouponServiceProtocol
 
 @available(iOS 17.0, *)
 private extension PointOfSaleCouponsController {
+    func loadFirstPage() async {
+        do {
+            // 1. Attempt to load the first page from local storage, and present them
+            let storedCoupons = try await couponProvider.provideLocalPointOfSaleCoupons()
+
+            if !storedCoupons.isEmpty {
+                setCouponsLoadedViewState(storedCoupons, hasMoreItems: true)
+            } else {
+                // 2. If there are no local results, retry again after syncing the first page from remote first
+                let coupons = try await couponProvider.providePointOfSaleCoupons(pageNumber: 1)
+                if !coupons.items.isEmpty {
+                    setCouponsLoadedViewState(coupons.items, hasMoreItems: true)
+                } else {
+                    setCouponsEmptyViewState()
+                }
+            }
+        } catch {
+            if let couponError = error as? PointOfSaleCouponServiceError {
+                setCouponsErrorViewState(couponError)
+            }
+        }
+    }
+
     @MainActor
-    func fetchCoupons(pageNumber: Int = Constants.firstPage) async {
+    func fetchCoupons(pageNumber: Int, appendToExistingCoupons: Bool = true) async throws -> Bool {
         do {
             let pagedCoupons = try await couponProvider.providePointOfSaleCoupons(pageNumber: pageNumber)
+
+            let newCoupons = pagedCoupons.items
+            var allCoupons = appendToExistingCoupons ? itemsViewState.itemsStack.root.items : []
+            let uniqueNewCoupons = newCoupons.filter { newCoupon in
+                !allCoupons.contains(newCoupon)
+            }
+            allCoupons.append(contentsOf: uniqueNewCoupons)
+
             let hasMoreItems = pagedCoupons.hasMorePages
             if pagedCoupons.items.isEmpty {
```

Comment:
Should this be all coupons? Same with `setCouponsLoadedViewState`. 

```suggestion
            if allCoupons.isEmpty {
```

Reply:
allCoupons are pagedCoupons here, as returns everything from the remote up to a given page, which makes it confusing with the unique coupons filtering. Updated on https://github.com/woocommerce/woocommerce-ios/pull/15499/commits/f265f8565aeab58375d26107935084c844cd1a62 , more in the comment below 👇